File: src/main/kotlin/jokatu/components/eventhandlers/AwaitingInputEventHandler.java
package jokatu.components.eventhandlers;

import jokatu.game.Game;
import jokatu.game.event.SpecificEventHandler;
import jokatu.game.input.AwaitingInputEvent;
import org.jetbrains.annotations.NotNull;
import org.springframework.stereotype.Component;

@Component
public class AwaitingInputEventHandler extends SpecificEventHandler<AwaitingInputEvent> {

	@NotNull
	@Override
	protected Class<AwaitingInputEvent> getEventClass() {
		return AwaitingInputEvent.class;
	}

	@Override
	protected void handleCastEvent(@NotNull Game<?> game, @NotNull AwaitingInputEvent event) {
		game.getPlayers().stream().forEach(
				player -> sender.sendToUser(
						player.getName(),
						"/topic/awaiting.game." + game.getIdentifier(),
						event.getAwaitingPlayers().contains(player)
				)
		);
	}
}

File: src/main/kotlin/jokatu/components/eventhandlers/AwaitingInputEventHandler.kt
package jokatu.components.eventhandlers

import jokatu.components.stomp.AwaitingInput
import jokatu.game.Game
import jokatu.game.event.SpecificEventHandler
import jokatu.game.input.AwaitingInputEvent
import org.springframework.stereotype.Component

@Component
class AwaitingInputEventHandler : SpecificEventHandler<AwaitingInputEvent>(AwaitingInputEvent::class) {
	override fun handleCastEvent(game: Game<*>, event: AwaitingInputEvent) {
		game.getPlayers().stream().forEach { player ->
			sender.sendToUser(player.name, AwaitingInput(game), event.awaitingPlayers.contains(player))
		}
	}
}
